lib/fluent/plugin/out_prometheus_pushgateway.rb

```diff
@@ -47,6 +70,21 @@ def configure(conf)
         super
 
         @push_client = ::Prometheus::Client::Push.new("#{@job_name}:#{fluentd_worker_id}", @instance, @gateway)
+
+        use_tls = gateway && (URI.parse(gateway).scheme == 'https')
+
+        if use_tls
+          # prometheus client doesn't have an interface to set the HTTPS options
+          http = @push_client.instance_variable_get(:@http)
```

Comment:
OMG!
Should we report this monkey patch as feature request to prometheus ruby client?

Reply:
I already did. https://github.com/prometheus/client_ruby/pull/181